### PR TITLE
feat: add ref map to resolver graph node data

### PR DESCRIPTION
**Pull request by lottamus**
**BREAKING CHANGE**: resolver.graph node data is now an object that has `data` and `refMap`.

```ts
export interface IGraphNodeData {
   /**
   * A map of the refs in this node, and where they point
   *
   * Example:
   *
   * ```json
   * {
   *   "#/users/address": "file:///api.json/#models/card",
   * }
   * ```
   */
  refMap: Dictionary<string>;
  data?: any;
}
```

### CODE REVIEW

**Comment by marbemac** [comment]
Chris and I met on this, and he's going to explore another option - so hold on any reviews!

**Comment by pytlesk4** [comment]
Curious as to why explore another option? @marbemac

**Comment by marbemac** [comment]
Basically the underlying issue here is that DepGraph does not support edges. It's simple single parent -> child relationship. What we actually need is a directed multigraph, which that other graphlib supports.

**Comment by pytlesk4** [comment]
Hmm, ok. I still think we are going to need to do something similar to what we were doing DepGraph. But we shall see.

File: src/types.ts [hunk 1]
```diff
@@ -222,6 +222,22 @@ export interface IRefHandlerOpts {
   parentPointer: string;
 }
 
+export interface IGraphNodeData {
```
Comment (pytlesk4) [inline src/types.ts]: Nice!

**Comment by marbemac** [comment]
@P0lip small breaking change here, that might affect that recent spectral PR? If so, I suggest we try to upgrade spectral to json-ref-resolver v3 asap. Should be a simple change.

**Pull request merged**